tuf/repository_tool.py

```diff
@@ -1777,7 +1778,64 @@ def __call__(self, rolename):
 
 
 
-  def add_delegated_role(self, rolename, targets_object):
+
+  def load_signing_key_succinct_delegations(self, key):
+    """
+    <Purpose>
+      Load the role key for all succinct delegations. All succinct delegations
+      use the same role key
+
+    <Arguments>
+      key:
+        The role key for all succinct delegations associated with this role,
+        conformant to 'securesystemslib.formats.ANYKEY_SCHEMA'.
+        It must contain the private key, so that role signatures may be
+        generated when writeall() or write() is eventually called to generate
+        valid metadata files.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError, if 'key' is improperly formatted.
+
+      securesystemslib.exceptions.Error, if the private key is not found in 'key'.
+
+    <Side Effects>
+      Updates the keydb and roledb entries for all succinct delegations
+
+    <Returns>
+      None.
+
+    """
+    # Does 'key' have the correct format?
+    # Ensure the arguments have the appropriate number of objects and object
+    # types, and that all dict keys are properly named.  Raise
+    # 'securesystemslib.exceptions.FormatError' if any are improperly formatted.
+    securesystemslib.formats.ANYKEY_SCHEMA.check_match(key)
+
+    # Ensure the private portion of the key is available, otherwise signatures
+    # cannot be generated when the metadata file is written to disk.
+    if 'private' not in key['keyval'] or not len(key['keyval']['private']):
+      raise securesystemslib.exceptions.Error('This is not a private key.')
+
+    # Has the key, with the private portion included, been added to the keydb?
+    # The public version of the key may have been previously added.
+    try:
+      tuf.keydb.add_key(key, repository_name=self._repository_name)
+
+    except tuf.exceptions.KeyAlreadyExistsError:
+      tuf.keydb.remove_key(key['keyid'], self._repository_name)
+      tuf.keydb.add_key(key, repository_name=self._repository_name)
```

Comment:
should we be handling a possible exception on the second `add_key` call here?

Reply:
I don't think so. If the KeyAlreadyExistsError is raised on the first call, the formatting and existence checks should have already passed.